cadc-vos-server/src/main/java/ca/nrc/cadc/vos/server/transfers/TransferRunner.java

```diff
@@ -182,9 +182,33 @@ private Transfer createTransfer(List<Parameter> params)
 
         Direction dir = new Direction(sdir);
 
-        Protocol proto = new Protocol(sproto);
         List<Protocol> plist = new ArrayList<Protocol>();
-        plist.add(proto);
+        plist.add(new Protocol(sproto));
+        
+        // also add a protocol with current securityMethod
+        AuthMethod am = AuthenticationUtil.getAuthMethod(AuthenticationUtil.getCurrentSubject());
+        Protocol proto = new Protocol(sproto);
+        if (am != null) {
+            switch (am) {
+                case CERT:
+                    proto.setSecurityMethod(Standards.SECURITY_METHOD_CERT);
+                    break;
+                case COOKIE:
+                    proto.setSecurityMethod(Standards.SECURITY_METHOD_COOKIE);
+                    break;
+                case PASSWORD:
+                    proto.setSecurityMethod(Standards.SECURITY_METHOD_HTTP_BASIC);
+                    break;
+                case TOKEN:
+                    proto.setSecurityMethod(Standards.SECURITY_METHOD_TOKEN);
+                    break;
+                default:
+                    break;
+            }
+        }
+        if (proto.getSecurityMethod() != null) {
+            plist.add(proto);
+        }
```

Comment:
Is this sequence to help 2.0 clients by adding the most likely intended security method?

Reply:
Yes, it explicitly adds a second protocol with securityMethod that matches the current AuthMethod since the client can't specify securityMethod via param but the service might implement strict 2.1 matching (like cavern does).